crates/wasi/src/snapshots/preview_1.rs

```diff
@@ -0,0 +1,151 @@
+    Ok(())
+    }
+}
+}
+
+impl_add_to_linker_for_funcs!(
+    args_get (arg0: i32, arg1: i32) => i32,
+    args_sizes_get(arg0: i32, arg1: i32) => i32,
+    environ_get (arg0: i32, arg1: i32) => i32,
+    environ_sizes_get (arg0: i32, arg1: i32) => i32,
+    clock_res_get (arg0: i32, arg1: i32) => i32,
+    clock_time_get (arg0 : i32, arg1 : i64, arg2 : i32) => i32,
+    fd_advise(arg0: i32, arg1: i64, arg2: i64, arg3: i32) => i32,
+    fd_allocate(arg0: i32, arg1: i64, arg2: i64) => i32,
+    fd_close(arg0: i32,) => i32,
+    fd_datasync(arg0: i32,) => i32,
+    fd_fdstat_get(arg0: i32, arg1: i32) => i32,
+    fd_fdstat_set_flags(arg0: i32, arg1: i32) => i32,
+    fd_fdstat_set_rights(arg0: i32, arg1: i64, arg2: i64) => i32,
+    fd_filestat_get(arg0: i32, arg1: i32) => i32,
+    fd_filestat_set_size(arg0: i32, arg1: i64) => i32,
+    fd_filestat_set_times(arg0: i32, arg1: i64, arg2: i64, arg3: i32) => i32,
+    fd_pread(arg0: i32, arg1: i32, arg2: i32, arg3: i64, arg4: i32) => i32,
+    fd_prestat_get(arg0: i32, arg1: i32) => i32,
+    fd_prestat_dir_name(arg0: i32, arg1: i32, arg2: i32) => i32,
+    fd_pwrite(arg0: i32, arg1: i32, arg2: i32, arg3: i64, arg4: i32) => i32,
+
+    fd_read(arg0: i32, arg1: i32, arg2: i32, arg3: i32) => i32,
+    fd_readdir(arg0: i32, arg1: i32, arg2: i32, arg3: i64, arg4: i32) => i32,
+    fd_renumber(arg0: i32, arg1: i32) => i32,
+    fd_seek(arg0: i32, arg1: i64, arg2: i32, arg3: i32) => i32,
+    fd_sync(arg0: i32) => i32,
+    fd_tell(arg0: i32, arg1: i32) => i32,
+    fd_write(arg0: i32, arg1: i32, arg2: i32, arg3: i32) => i32,
+    path_create_directory(arg0: i32, arg1: i32, arg2: i32) => i32,
+    path_filestat_get(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32) => i32,
+    path_filestat_set_times(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i64, arg5: i64, arg6: i32) => i32,
+    path_link(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32, arg5: i32, arg6: i32) => i32,
+    path_open(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32, arg5: i64, arg6: i64, arg7: i32, arg8: i32) => i32,
+    path_readlink(arg0: i32, rg1: i32, rg2: i32, rg3: i32, rg4: i32, rg5: i32) => i32,
+    path_remove_directory(arg0: i32, arg1: i32, arg2: i32) => i32,
+    path_rename(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32, arg5: i32) => i32,
+    path_symlink(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32) => i32,
+    path_unlink_file(arg0: i32, arg1: i32, arg2: i32) => i32,
+    poll_oneoff(arg0: i32, arg1: i32, arg2: i32, arg3: i32) => i32,
+    proc_raise(arg0: i32) => i32,
+    proc_exit(arg0: i32) => (),
+    sched_yield() => i32,
+    random_get(arg0: i32, arg1: i32) => i32,
+    sock_accept(arg0: i32, arg1: i32, arg2: i32) => i32,
+    sock_recv(arg0: i32, arg1: i32, arg2: i32, arg3: i32, arg4: i32, arg5: i32) => i32,
+    sock_send(arg0: i32,arg1: i32,arg2: i32,arg3: i32,arg4: i32) => i32,
+    sock_shutdown(arg0: i32, arg1: i32) => i32,
+
+);
+
+pub fn define_wasi<T, U>(
```

Comment:
Requires docs since part of public API.

Reply:
Sure.

Thanks. I now see why that CI command failed.